src/util/irep.h

```diff
@@ -344,57 +387,29 @@ typename sharing_treet<derivedt>::dt sharing_treet<derivedt>::empty_d;
 /// code / one semi-colon). The most common descendant of \ref codet is
 /// \ref code_assignt so a common pattern is to cast the \ref codet to an
 /// assignment and then recurse on the expression on either side.
+class irept
 #ifdef SHARING
-class irept : public sharing_treet<irept>
+  : public sharing_treet<irept>
 #else
-class irept
+  : public non_sharing_treet<irept>
 #endif
 {
 public:
-#ifndef SHARING
-  // These are not stable.
-  typedef std::vector<irept> subt;
-
-  typedef tree_nodet<irept, false> dt;
-
-  typedef dt::named_subt named_subt;
-#endif
-  // named_subt has to provide stable references; with C++11 we could
-  // use std::forward_list or std::vector< unique_ptr<T> > to save
-  // memory and increase efficiency.
-
+  using supert = tree_implementationt;
```

Comment:
I think it would be slightly more consistent to call it `baset` rather than `supert`. `baset` is used in various other places in the codebase.